pkg/operator/starter.go

```diff
@@ -50,62 +52,95 @@ const (
 )
 
-func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext) error {
-	// Create core clientset and informer
-	kubeClient := kubeclient.NewForConfigOrDie(rest.AddUserAgent(controllerConfig.KubeConfig, operatorName))
-	kubeInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(kubeClient, defaultNamespace, cloudConfigNamespace, "")
-	secretInformer := kubeInformersForNamespaces.InformersFor(defaultNamespace).Core().V1().Secrets()
-	nodeInformer := kubeInformersForNamespaces.InformersFor("").Core().V1().Nodes()
-	configMapInformer := kubeInformersForNamespaces.InformersFor(defaultNamespace).Core().V1().ConfigMaps()
-
-	// Create config clientset and informer. This is used to get the cluster ID
-	configClient := configclient.NewForConfigOrDie(rest.AddUserAgent(controllerConfig.KubeConfig, operatorName))
-	configInformers := configinformers.NewSharedInformerFactory(configClient, 20*time.Minute)
-	infraInformer := configInformers.Config().V1().Infrastructures()
-
-	// Create informer for the ConfigMaps in the openshift-config-managed namespace. This is used to get the custom CA
-	// bundle to use when accessing the AWS API.
-	cloudConfigInformer := kubeInformersForNamespaces.InformersFor(cloudConfigNamespace).Core().V1().ConfigMaps()
-	cloudConfigLister := cloudConfigInformer.Lister().ConfigMaps(cloudConfigNamespace)
-
-	// Create GenericOperatorclient. This is used by the library-go controllers created down below
-	gvr := opv1.SchemeGroupVersion.WithResource("clustercsidrivers")
-	operatorClient, dynamicInformers, err := goc.NewClusterScopedOperatorClientWithConfigName(controllerConfig.KubeConfig, gvr, instanceName)
+func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext, guestKubeConfigString *string) error {
+	// Create core clientset and informer for the MANAGEMENT cluster.
+	eventRecorder := controllerConfig.EventRecorder
+	controlPlaneNamespace := controllerConfig.OperatorNamespace
+	controlPlaneKubeClient := kubeclient.NewForConfigOrDie(rest.AddUserAgent(controllerConfig.KubeConfig, operatorName))
+	controlPlaneKubeInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(controlPlaneKubeClient, controlPlaneNamespace, cloudConfigNamespace, "")
+	controlPlaneSecretInformer := controlPlaneKubeInformersForNamespaces.InformersFor(controlPlaneNamespace).Core().V1().Secrets()
+	controlPlaneConfigMapInformer := controlPlaneKubeInformersForNamespaces.InformersFor(controlPlaneNamespace).Core().V1().ConfigMaps()
+
+	// Create informer for the ConfigMaps in the openshift-config-managed namespace.
+	// This is used to get the custom CA bundle to use when accessing the AWS API.
+	// This is only used in standalone OCP clusters.
+	controlPlaneCloudConfigInformer := controlPlaneKubeInformersForNamespaces.InformersFor(controlPlaneNamespace).Core().V1().ConfigMaps()
+	controlPlaneCloudConfigLister := controlPlaneCloudConfigInformer.Lister().ConfigMaps(controlPlaneNamespace)
```

Comment:
The comment above says it should be `cloudConfigNamespace`, but I can see `controlPlaneNamespace` is used.

Should it use `cloudConfigNamespace` in standalone OCP and `controlPlaneNamespace` in HyperShift?

Reply:
Good catch. IMO our controller should react to changes in the `kube-cloud-config` ConfigMap located in the same namespace the operator is on (i.e., `controlPlaneNamespace`), because that's the ConfigMap used to get the custom CA bundle.

That ConfigMap is synced from `openshift-config-managed/kube-cloud-config` by `caSyncController`, and **that** controller should be the one watching the `openshift-config-managed` namespace.

I fixed the comment.

Thoughts?

Reply:
lgtm